api/v1alpha1/externaldns_webhook.go

```diff
@@ -129,3 +148,21 @@ func (r *ExternalDNS) validateProviderCredentials() error {
 	}
 	return nil
 }
+
+// Returns Openshift Container Platform Major Version
+func GetOCPMajorVersion(kubeClient discovery.DiscoveryInterface) int {
```

Comment:
I think that the function name doesn't reflect what it really does. For example, major version 3 will never be returned.
It may be more logical to name the function to something binary: `IsOCP4` or similar.

Reply:
Actually OCP4 name will be specific 4th version. So, I think in the future we may have to add 5 as major version.

Reply:
Yes, you are right, my example was too specific to OCP4.
What I wanted to say was more about the nature of the function. It's binary (yes/no answer) by nature for the moment, it doesn't return version 3 or won't return version 5 when we have it out, it just checks the presence of the OCP API resource.